src/daft-dsl/src/functions/scalar.rs

```diff
@@ -84,3 +85,337 @@ impl Display for ScalarFunction {
         Ok(())
     }
 }
+
+/// generates a scalar function implementation
+/// # Example
+/// ```rust,ignore
+///
+/// make_unary_udf_function!{
+///     name: "my_udf",
+///     to_field: (input0, schema) {
+///       // implementation
+///     },
+///     evaluate: (input0) {
+///       // implementation
+///     }
+/// }
+/// ```
+/// generates the following code
+/// ```rust,ignore
+/// #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, PartialEq, Eq, Hash)]
+/// pub struct MyUDF {}
+///
+/// #[typetag::serde]
+/// impl ScalarUDF for MyUDF {
+///    fn as_any(&self) -> &dyn std::any::Any { self }
+///    fn name(&self) -> &'static str { "my_udf" }
+///
+///    fn to_field(&self, inputs: &[ExprRef], schema: &daft_core::schema::Schema) -> common_error::DaftResult<daft_core::datatypes::Field> {
+///      match inputs {
+///        [input0] => { // implementation }
+///        _ => Err(common_error::DaftError::InvalidInput("my_udf".to_string()))
+///      }
+///    }
+///
+///    fn evaluate(&self, inputs: &[daft_core::series::Series]) -> common_error::DaftResult<daft_core::series::Series> {
+///      match inputs {
+///        [input0] => { // implementation }
+///        _ => Err(common_error::DaftError::InvalidInput("my_udf".to_string()))
+///      }
+///    }
+/// }
+/// fn my_udf(input: ExprRef) -> ExprRef {
+///    ScalarFunction::new(MyUDF {}, vec![input]).into()
+/// }
+///
+/// #[cfg(feature = "python")]
+/// #[pyo3::pyfunction]
+/// #[pyo3(name = "my_udf")]
+/// pub fn py_my_udf(expr: PyExpr) -> PyResult<PyExpr> {
+///   my_udf(expr.into()).into()
+/// }
+/// ```
+/// These are not meant to cover all cases, but to provide a starting point for implementing most of the basic functions
+#[macro_export]
+macro_rules! make_udf_function {
```

Comment:
I see that we use `make_udf_function` a lot inside of `daft-functions`. Why not just define it there instead of in `daft-dsl`? That way, we could avoid the `#[macro_export]`. Is `make_udf_function` used inside of `daft-dsl` as well? I may have missed those occurrences.